test/integration/connect/envoy/case-cross-peers/primary/service_s1.hcl

```diff
@@ -9,6 +9,9 @@ services {
             destination_name = "s2"
             destination_peer = "primary-to-alpha"
             local_bind_port  = 5000
+            mesh_gateway {
```

Comment:
Using local mode on the existing case tests both code paths because the local mesh gateway just forwards requests to the remote one. Does it make sense to test both cases separately? I left just one case because it feels more consistent with our existing integration tests.

Reply:
Seems OK to me.